### PR TITLE
`IndexHierachy.from_values_per_depth()`

### PR DESCRIPTION
**Pull request by flexatone**

### CODE REVIEW

File: static_frame/core/index_hierarchy.py [hunk 1]
```diff
@@ -447,22 +447,36 @@ def _from_arrays(cls: tp.Type[IH],
...
+                    raise ErrorInitIndex('per depth iterables must be the same length')
+
             # NOTE: we are not checking that they are all 1D
             depth = len(arrays)
             column_iter = arrays
 
         if not size:
-            return cls._from_empty((), name=name, depth_reference=depth_reference)
+            if depth is None:
+                raise RuntimeError('depth_reference must be specified for empty values')
```
Comment (chaburkland) [inline static_frame/core/index_hierarchy.py:478]: Thanks for catching this!

File: static_frame/core/index_hierarchy.py [hunk 2]
```diff
@@ -474,17 +488,24 @@ def _from_arrays(cls: tp.Type[IH],
...
 
-        blocks = TypeBlocks.from_blocks(arrays)
 
         return cls(
                 indices=indices,
                 indexers=indexers,
                 name=name,
                 blocks=blocks,
-                own_blocks=True,
+                own_blocks=own_blocks,
```
Comment (chaburkland) [inline static_frame/core/index_hierarchy.py:508]: `own_blocks` is always `True` per the above branch

**Review by chaburkland — APPROVED** [review]
Awesome! Thanks for extending this

File: static_frame/core/index_hierarchy.py (hunk 2 above)
Comment (flexatone) [reply static_frame/core/index_hierarchy.py:508]: Good catch; it should be False on 497!

Comment (chaburkland) [reply static_frame/core/index_hierarchy.py:508]: @flexatone Did you fix this? line 497 still says `own_blocks=True`?

Comment (flexatone) [reply static_frame/core/index_hierarchy.py:508]: Thanks for reminding me, I am surprised it did not result in an error!

**Pull request merged**